Products/CMFPlone/CatalogTool.py

```diff
@@ -425,6 +427,21 @@ def indexObject(obj, path):
                     safe_callable(obj.indexObject)):
                 try:
                     obj.indexObject()
+
```

Comment:
a few things about this implementation:

1. use the api: `api.portal.get_tool('portal_catalog')` and move that line inside the `if` block
2. is possible to be in here without access to the catalog?
3. when exactly that `StopIteration` could happen? is not obvious to me